runtime/service.go

```diff
@@ -514,14 +522,22 @@ func (s *service) StopVM(requestCtx context.Context, request *proto.StopVMReques
 		return nil, err
 	}
 
-	// The graceful shutdown logic, including stopping the VM, is centralized in Shutdown. We set "Now" to true
-	// to ensure that the service will actually shutdown even if there are still containers being managed.
-	_, err = s.Shutdown(requestCtx, &taskAPI.ShutdownRequest{Now: true})
-	if err != nil {
-		return nil, err
-	}
+	shutdownCh := make(chan error)
+	go func() {
+		defer close(shutdownCh)
+		_, err := s.Shutdown(requestCtx, &taskAPI.ShutdownRequest{Now: true})
+		shutdownCh <- err
+	}()
 
-	return &empty.Empty{}, nil
+	select {
+	case <-timer.C:
+		return nil, status.Error(codes.DeadlineExceeded, "timed out waiting for VM shutdown")
```

Comment:
In this case do we need err to be set to != nil so that shimCancel is called in the deferred func above?

Reply:
It's not straightforward, but that is what's happening implicitly here: https://play.golang.org/p/xqGlTUUX8jq

Reply:
ah I missed that it had a named return for err. Thanks!